backend/setupNunjucks.js

```diff
@@ -10,6 +13,131 @@ module.exports = (app) => {
     }
   )
 
+  njkEnv.addFilter('findError', (array, formFieldId) => {
+    if (!array) return null
+    const item = array.find((error) => error.href === `#${formFieldId}`)
+    if (item) {
+      return {
+        text: item.text,
+      }
+    }
+    return null
+  })
+
+  njkEnv.addFilter('findErrors', (errors, formFieldIds) => {
+    if (!errors) return null
+    const fieldIds = formFieldIds.map((field) => `#${field}`)
+    const errorIds = errors.map((error) => error.href)
+    const firstPresentFieldError = fieldIds.find((fieldId) => errorIds.includes(fieldId))
+    if (firstPresentFieldError) {
+      return { text: errors.find((error) => error.href === firstPresentFieldError).text }
+    }
+    return null
+  })
+
+  njkEnv.addFilter('formatDate', (value, format) => {
+    return value ? moment(value).format(format) : null
+  })
+
+  njkEnv.addFilter('hasErrorWithPrefix', (errorsArray, prefixes) => {
+    if (!errorsArray) return null
+    const formattedPrefixes = prefixes.map((field) => `#${field}`)
+    return errorsArray.some((error) => formattedPrefixes.some((prefix) => error.href.startsWith(prefix)))
+  })
+
+  njkEnv.addFilter('toTextValue', (array, selected) => {
+    if (!array) return null
+
+    const items = array.map((entry) => ({
+      text: entry,
+      value: entry,
+      selected: entry && entry === selected,
+    }))
+
+    return [
+      {
+        text: '--',
+        value: '',
+        hidden: true,
+        selected: true,
+      },
+      ...items,
+    ]
+  })
+
+  njkEnv.addFilter('addDefaultSelectedVale', (items, text, show) => {
```

Comment:
We still need to do a find replace on this in PSH, may as well fix it here before we start using it. `addDefaultSelectedVale` `value`